src/main/java/org/mafagafogigante/dungeon/commands/CommandLimits.java

```diff
@@ -0,0 +1,19 @@
+package org.mafagafogigante.dungeon.commands;
+
+import org.jetbrains.annotations.NotNull;
+
+class CommandLimits {
+
+  /**
+   * An arbitrary maximum command length.
+   *
+   * <p>This value is used for input validation and to prevent that slower and memory-consuming methods ever get feed
```

Comment:
If an inefficient implementation were to use a matrix of integers to compute distances on this input, it would take, at most, 4 * (2048)² bytes, which is 16 MiB, which is safe against OOM errors.